### PR TITLE
feat(@vtmn/svelte): add `VtmnNavbar` component

### PR DESCRIPTION
**Pull request by jb-thery**
## Changes description

This P.R adds the Svelte navbar component

## Context

Related issue [#740](https://github.com/Decathlon/vitamin-web/issues/740)

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch**. Please, don't request directly from your main!
- [x] Check commits & PR names matches our requested structure. It must follow the https://www.conventionalcommits.org pattern.
- [x] Check your code additions will fail neither code linting checks.
- [x] I have reviewed the submitted code.
- [x] I have tested on related showcases.
- [x] If it includes design changes, please ask for a review with a core team designer.

## Does this introduce a breaking change?

- No

### CODE REVIEW

**Comment by Tlahey** [comment]
@jb-thery good PR but I think the navbar component are too specific.
The component must not linked to decathlon.

Here the best way (I think) is to provide slots under the component for help users to set items inside.

image

So, let's create under the navbar 4 slots like this 
```svelte 
<nav class="vtmn-navbar" {...$$restProps}>
  {#if $$slots['left-navigation'] }
    <div class="vtmn-navbar_left-navigation">
      <slot name="left-navigation" />
    </div>
  {/if}
  {#if $$slots['navbar-logo'] }
    <a class="vtmn-navbar_logo" href="#">
      <slot name="navbar-logo" />
    </a>
  {/if}
  {#if $$slots['middle-area'] }
    <div class="vtmn-navbar_middle-area">
      <slot name="middle-area" />
    </div>
  {/if}
  {#if $$slots['right-navigation'] }
    <div class="vtmn-navbar_right-navigation">
      <slot name="right-navigation" />
    </div>
  {/if}
</nav>
```

**Review by lauthieb — COMMENTED** [review]
Thanks @jb-thery for your contribution!!
I will wait for #1027 before reviewing your PR because it's a dependency and it could be cool to directly use this component inside the `VtmnNavbar` thanks a lot!

**Review by lauthieb — CHANGES_REQUESTED** [review]
As mentioned in my previous comment, for the moment I'm waiting for #1027 before reviewing your PR.
But agree with @Tlahey to be the most generic as possible for this component like we did on Figma. 👍

File: packages/sources/svelte/src/components/navigation/VtmnNavbar/VtmnNavbar.svelte [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+  /**
+   * Delete slot div from de DOM if slot doesn't exist.
+   *
+   * @param {'left-nav' | 'logo' | 'middle-area' | 'right-nav'} slotName
+   * @returns {boolean} slotExists
+   */
+  const checkSlotExists = (slotName) => {
+    return SLOTS && SLOTS[slotName] && SLOTS[slotName].length;
+  };
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/navigation/VtmnNavbar/VtmnNavbar.svelte:25]: You can just use `$$slots.name` 
```
if ($$slots[slotName]) {
    // Do stuff
}
```

File: packages/showcases/svelte/stories/components/navigation/VtmnSearch/VtmnSearch.stories.svelte [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  {argTypes}
+  {parameters}
+/>
+
+<Template let:args>
+  <VtmnSearch
+    {...args}
+    on:input={(e) => console.log(e.target.value)}
+    on:search={(event) => alert('Search event fired ! :' + event.detail.text)}
+  />
```
Comment (Tlahey) [inline packages/showcases/svelte/stories/components/navigation/VtmnSearch/VtmnSearch.stories.svelte:22]: Please add slots on the template to match the css example :)

File: packages/showcases/svelte/stories/components/navigation/VtmnNavbar/VtmnNavbar.stories.svelte [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  component={VtmnNavbar}
+  {argTypes}
+  {parameters}
+/>
+
+<Story name="Overview" let:args>
+  <div style="width: 800px; display: flex; justify-content: center">
+    <VtmnNavbar {...args}>
+      <svelte:fragment slot="logo">
+        <a class="vtmn-navbar_logo" href="#">
```
Comment (Tlahey) [inline packages/showcases/svelte/stories/components/navigation/VtmnNavbar/VtmnNavbar.stories.svelte:21]: Question : 
I think the `<a class="vtmn-navbar_logo" href="#">` should be inside the component `VtmnNavbar`.

@lauthieb , for all logo, if we click on it, will it always have to go back to the href '#' ?

File: packages/showcases/svelte/stories/components/navigation/VtmnNavbar/VtmnNavbar.stories.svelte [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      </svelte:fragment>
+    </VtmnNavbar>
+  </div>
+</Story>
+
+<Story name="With menu and cta" let:args>
+  <div style="width: 800px; display: flex; justify-content: center">
+    <VtmnNavbar {...args}>
+      <svelte:fragment slot="left-nav">
+        <button
```
Comment (Tlahey) [inline packages/showcases/svelte/stories/components/navigation/VtmnNavbar/VtmnNavbar.stories.svelte:55]: Use `VtmnButton` component

File: packages/showcases/svelte/stories/components/navigation/VtmnNavbar/VtmnNavbar.stories.svelte [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+      </svelte:fragment>
+    </VtmnNavbar>
+  </div>
+</Story>
+
+<Story name="With menu and button" let:args>
+  <div style="width: 800px; display: flex; justify-content: center">
+    <VtmnNavbar {...args}>
+      <svelte:fragment slot="left-nav">
+        <button
```
Comment (Tlahey) [inline packages/showcases/svelte/stories/components/navigation/VtmnNavbar/VtmnNavbar.stories.svelte:110]: `VtmnButton` :)

File: packages/sources/svelte/src/components/navigation/VtmnNavbar/VtmnNavbar.svelte [hunk 6]
```diff
@@ -0,0 +1,56 @@
+<script>
+  import { cn } from '../../../utils/classnames';
+
+  /** @restProps { button } */
+
+  const SLOTS = $$props.$$slots;
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/navigation/VtmnNavbar/VtmnNavbar.svelte:6]: remove

File: packages/sources/svelte/src/components/navigation/VtmnNavbar/VtmnNavbar.svelte [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+
+  /**
+   * Custom classes to apply to the component.
+   *
+   * @type {string}
+   */
+  export let className = '';
```
Comment (Tlahey) [inline packages/sources/svelte/src/components/navigation/VtmnNavbar/VtmnNavbar.svelte:13]:
```
  let className = '';
  /**
   * @type {string} Custom classes to apply to the component.
   */
  export { className as class };
```

File: packages/showcases/svelte/stories/components/navigation/VtmnNavbar/VtmnNavbar.stories.svelte (hunk 3 above)
Comment (jb-thery) [reply packages/showcases/svelte/stories/components/navigation/VtmnNavbar/VtmnNavbar.stories.svelte:21]: @lauthieb what do you think about it ?

Comment (lauthieb) [reply packages/showcases/svelte/stories/components/navigation/VtmnNavbar/VtmnNavbar.stories.svelte:21]: @Tlahey @jb-thery totally agree. Logo should be inside this component and it will always redirect to the homepage, so I think "#" in `href` is correct.

File: packages/sources/svelte/src/components/navigation/VtmnNavbar/VtmnNavbar.svelte [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  import { cn } from '../../../utils/classnames';
+
+  /** @restProps { button } */
+
+  /**
+   * Logo anchor.
+   *
+   * @type {string} [href]
+   */
+  export let href = '#';
```
Comment (lauthieb) [inline packages/sources/svelte/src/components/navigation/VtmnNavbar/VtmnNavbar.svelte:11]: It could be better to rename it `logoHref`

File: packages/sources/svelte/rollup.config.js [hunk 9]
```diff
@@ -22,6 +22,10 @@ const src = {
       folder: 'navigation',
       components: ['VtmnSearch'],
     },
+    {
+      folder: 'navigation',
+      components: ['VtmnSearch', 'VtmnNavbar'],
+    },
```
Comment (lauthieb) [inline packages/sources/svelte/rollup.config.js:28]: We always have this line just above (line 23), you don't need to add this new object.

File: packages/sources/svelte/package.json [hunk 10]
```diff
@@ -43,6 +43,8 @@
     "@vtmn/css-divider": "^0.1.21",
     "@vtmn/css-link": "^0.6.15",
     "@vtmn/css-list": "^0.2.17",
+    "@vtmn/css-navbar": "^0.1.19",
+    "@vtmn/css-search": "^0.4.11",
```
Comment (lauthieb) [inline packages/sources/svelte/package.json:47]: why do you have two imports?

File: packages/showcases/core/csf/components/navigation/search.csf.js [hunk 11]
```diff
@@ -10,45 +10,27 @@ export const parameters = {
...
-    description: 'The placeholder of the search bar.',
-    defaultValue: 'Search',
-    control: { type: 'text' },
-  },
-  value: {
-    type: { name: 'string', required: false },
-    description: 'The value of the search bar.',
-    defaultValue: undefined,
-    control: { type: 'text' },
-  },
```
Comment (lauthieb) [inline packages/showcases/core/csf/components/navigation/search.csf.js:53]: I think `@vtmn/react` needs this. How can we resolve this?

File: packages/showcases/core/csf/components/navigation/navbar.csf.js [hunk 12]
```diff
@@ -1,9 +1,21 @@
...
+    type: { name: 'string', required: false },
+    description: 'Logo anchor',
+    defaultValue: '#',
+    control: { type: 'text' },
+  },
+  class: {
+    type: { name: 'string', required: false },
+    description: 'Custom classes to apply to the component',
+    defaultValue: false,
+    control: { type: 'text' },
```
Comment (lauthieb) [inline packages/showcases/core/csf/components/navigation/navbar.csf.js:19]: Why did you add "class"?

File: packages/showcases/core/csf/components/navigation/navbar.csf.js [hunk 13]
```diff
@@ -1,9 +1,21 @@
...
-  },
   design: {
     type: 'figma',
-    url: 'https://www.figma.com/file/zDZIyayUlr1yTWrsi7cFoo/?node-id=4665%3A10809',
+    url: 'https://www.figma.com/file/zDZIyayUlr1yTWrsi7cFoo/Vitamin---Web-UI-kit-(v0)?node-id=10847%3A27832',
+  },
+};
+
+export const argTypes = {
+  href: {
```
Comment (lauthieb) [inline packages/showcases/core/csf/components/navigation/navbar.csf.js:9]: Should be renamed to `logoHref` I think it could be better.

**Review by lauthieb — CHANGES_REQUESTED** [review]
Thanks @jb-thery, just few remarks :)

File: packages/showcases/core/csf/components/navigation/search.csf.js [hunk 14]
```diff
@@ -10,45 +10,27 @@ export const parameters = {
 
 export const argTypes = {
   variant: {
-    type: { name: 'string', required: false },
-    description: 'The variant of the search bar.',
-    defaultValue: 'default',
+    type: { name: 'string', required: true },
+    description: 'Color variant of search component.',
```
Comment (GaspardMathon) [inline packages/showcases/core/csf/components/navigation/search.csf.js:14]:
```suggestion
    description: 'Variant of search component.',
```

**Review by GaspardMathon — COMMENTED** [review]
Few changes to make
